bin/generateSqlLite

```diff
@@ -383,52 +383,58 @@ try {
-                $stmt->bindParam(':host_host_id', $value['host_host_id'], PDO::PARAM_INT);                
-                $stmt->execute();	
+                $stmt->bindParam(':host_host_id', $value['host_host_id'], PDO::PARAM_INT);
+                $stmt->execute();
             }
 
             // Insert direct services
-            $insertedServices = array();            
-            foreach ($result_host_services as $value) {
-                if (!isset($insertedServices[$value['service_id']])) {
+            $insertedServices = [];
+            foreach ($resultHostServices as $service) {
+                if (!isset($insertedServices[$service['service_id']])) {
+
                     $stmt = $dbh_sqlite->prepare("INSERT INTO service (service_id, service_description, service_template_model_stm_id) VALUES (
                         :service_id, :service_description, :service_template_model_stm_id)");
-                    $stmt->bindParam(':service_id', $value['service_id'], PDO::PARAM_INT);
-                    $stmt->bindParam(':service_description', $value['service_description'], PDO::PARAM_STR);
-                    $stmt->bindParam(':service_template_model_stm_id', $value['service_template_model_stm_id'], PDO::PARAM_INT);
-                    $stmt->execute();
-                    $insertedServices[$value['service_id']] = true;
-                
-                    $stmt = $dbh_sqlite->prepare("INSERT INTO host_service_relation (service_service_id, host_host_id) VALUES (
-                        :service_service_id, :host_host_id)");
-                    $stmt->bindParam(':service_service_id', $value['service_id'], PDO::PARAM_INT);
-                    $stmt->bindParam(':host_host_id', $value['host_id'], PDO::PARAM_INT);
+                    $stmt->bindParam(':service_id', $service['service_id'], PDO::PARAM_INT);
+                    $stmt->bindParam(':service_description', $service['service_description'], PDO::PARAM_STR);
+                    $stmt->bindParam(':service_template_model_stm_id', $service['service_template_model_stm_id'], PDO::PARAM_INT);
                     $stmt->execute();
 
+                    $insertedServices[$service['service_id']] = true;
+
                     $stmt = $dbh_sqlite->prepare("INSERT INTO extended_service_information (service_service_id, esi_notes) VALUES (
                         :service_service_id, :esi_notes)");
-                    $stmt->bindParam(':service_service_id', $value['service_id'], PDO::PARAM_INT);
-                    $stmt->bindParam(':esi_notes', $value['esi_notes'], PDO::PARAM_STR);
+                    $stmt->bindParam(':service_service_id', $service['service_id'], PDO::PARAM_INT);
+                    $stmt->bindParam(':esi_notes', $service['esi_notes'], PDO::PARAM_STR);
                     $stmt->execute();
                 }
             }
+
+            //match hosts with their dedicated service
+            foreach ($resultHostServices as $service) {
```

Comment:
why looping twice on $resultHostServices ?

Reply:
Because before the changes we could only insert values to the table host_service_relation once because of the condition at line 400. I added a second loop to match all hosts with their service.

Reply:
file updated